### PR TITLE
Update data model to include refundedat

### PR DESCRIPTION
**Pull request by Jideboris**
__Short description outlining key changes/additions__

ROE-1469

### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/go.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/golang/go/wiki/CodeReviewComments)__

### CODE REVIEW

File: models/refund_db.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package models
 // RefundResourceDB represents the database refund structure
 type RefundResourceDB struct {
 	RefundId          string `bson:"refund_id"`
+	RefundedAt        string `bson:"refunded_at"`
```
Comment (emead) [inline models/refund_db.go:6]: I think this should be `time.Time` for consistency with other database items
```suggestion
	RefundedAt        time.Time `bson:"refunded_at"`
```

File: dao/mongo.go [hunk 2]
```diff
@@ -333,7 +333,7 @@ func (m *MongoService) PatchPaymentsWithRefundPendingStatus(id string, isPaid bo
 		patchUpdate = bson.M{
 			"$set": bson.M{
 				"refunds.$[x].status":      "refund-success",
-				"refunds.$[x].refunded_at": time.Now(),
+				"refunds.$[x].refunded_at": time.Now().String(),
```
Comment (emead) [inline dao/mongo.go:336]: Shouldn't need to convert to string if item is time.Time (see other comment)

File: models/refund_rest.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ type RefundResponse struct {
 // RefundResourceRest is the data contained in a refund resource
 type RefundResourceRest struct {
 	RefundId          string `json:"refund_id"`
+	RefundedAt        string `json:"refunded_at,omitempty"`
```
Comment (emead) [inline models/refund_rest.go:20]: Should also be time.Time

File: models/refund_db.go (hunk 1 above)
Comment (Jideboris) [reply models/refund_db.go]: Fixed

File: models/refund_rest.go (hunk 3 above)
Comment (Jideboris) [reply models/refund_rest.go]: fixed

**Review by emead — APPROVED** [review]

**Pull request merged**